js/ui/codingManager.js

```diff
@@ -84,7 +62,16 @@ const CodingManager = new Lang.Class({
             return false;
 
         let window = actor.meta_window;
-        if (!this._isBuilder(window.get_flatpak_id()))
+        let tracker = Shell.WindowTracker.get_default();
+        let correspondingApp = tracker.get_window_app(window);
+        let isSpeedwagonForBuilder = (
+            Shell.WindowTracker.is_speedwagon_window(window) &&
+            correspondingApp &&
+            correspondingApp.get_id() === 'org.gnome.Builder.desktop'
+        );
```

Comment:
These lines are repeated below; you can refactor them out in a `_isBuilderSpeedwagon()` function.